iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/ClearTextProtocolDowngradeCheck.java

```diff
@@ -74,47 +72,55 @@ public class ClearTextProtocolDowngradeCheck implements IacCheck {
     .with(CURL_COMMAND)
     .withPredicatesFrom(REDIRECTION_PREDICATES)
     .withPredicatesFrom(PROTO_FLAG_MISSING_OPTION_PREDICATES)
-    .with(SENSITIVE_BEGINNING_URL_SCHEME)
+    .with(SENSITIVE_HTTPS_URL_BEGINNING)
     .build();
 
   // matching "curl --proto -L https://redirecttoinsecure.example.com"
   private static final CommandDetector SENSITIVE_CURL_COMMAND_FLAG_WITH_MISSING_OPTION_DIFF_ORDER = CommandDetector.builder()
     .with(CURL_COMMAND)
     .withPredicatesFrom(PROTO_FLAG_MISSING_OPTION_PREDICATES)
     .withPredicatesFrom(REDIRECTION_PREDICATES)
-    .with(SENSITIVE_BEGINNING_URL_SCHEME)
+    .with(SENSITIVE_HTTPS_URL_BEGINNING)
     .build();
 
   // matching "curl -L https://redirecttoinsecure.example.com"
   private static final CommandDetector SENSITIVE_CURL_COMMAND_MISSING_FLAG = CommandDetector.builder()
     .with(CURL_COMMAND)
     .withPredicatesFrom(REDIRECTION_PREDICATES)
     .withPredicatesFrom(PROTO_FLAG_MISSING_PREDICATES)
-    .with(SENSITIVE_BEGINNING_URL_SCHEME)
+    .with(SENSITIVE_HTTPS_URL_BEGINNING)
     .build();
 
   // matching "curl -L --proto =foo https://redirecttoinsecure.example.com"
   private static final CommandDetector SENSITIVE_CURL_COMMAND_FLAG_WITH_WRONG_OPTION = CommandDetector.builder()
     .with(CURL_COMMAND)
     .withPredicatesFrom(REDIRECTION_PREDICATES)
     .withPredicatesFrom(PROTO_FLAG_WITH_WRONG_OPTION_PREDICATES)
-    .with(SENSITIVE_BEGINNING_URL_SCHEME)
+    .with(SENSITIVE_HTTPS_URL_BEGINNING)
     .build();
 
   // matching "curl --proto =foo -L https://redirecttoinsecure.example.com"
   private static final CommandDetector SENSITIVE_CURL_COMMAND_FLAG_WITH_WRONG_OPTION_DIFF_ORDER = CommandDetector.builder()
     .with(CURL_COMMAND)
     .withPredicatesFrom(PROTO_FLAG_WITH_WRONG_OPTION_PREDICATES)
     .withPredicatesFrom(REDIRECTION_PREDICATES)
-    .with(SENSITIVE_BEGINNING_URL_SCHEME)
+    .with(SENSITIVE_HTTPS_URL_BEGINNING)
+    .build();
+
+  private static final CommandDetector WGET_DETECTOR = CommandDetector.builder()
+    .with("wget")
+    .with(SENSITIVE_HTTPS_URL_BEGINNING)
+    .withOptionalRepeating(s -> s.startsWith("-") && !isCompliantRedirect(s))
```

Comment:
Maybe, `withAnyFlagExcept()` can also suit. That way you don't need to implement `isCompliantRedirect()`.